deviceshifu/pkg/deviceshifu/deviceshifu.go

```diff
@@ -180,11 +199,43 @@ func (handler DeviceCommandHandlerHTTP) commandHandleFunc() http.HandlerFunc {
 			}
 		}
 
-		log.Printf("handling instruction '%v' to '%v'", handlerInstruction, *handlerEdgeDeviceSpec.Address)
-		resp, err := handlerHTTPClient.Get("http://" + *handlerEdgeDeviceSpec.Address + "/" + handlerInstruction)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusServiceUnavailable)
-			log.Printf("HTTP error" + err.Error())
+		var resp *http.Response
+		var httpErr error
+		reqType := r.Method
+
+		log.Printf("handling instruction '%v' to '%v' with request type %v", handlerInstruction, *handlerEdgeDeviceSpec.Address, reqType)
+
+		if reqType == http.MethodGet {
+			httpUri := createUriFromRequest(*handlerEdgeDeviceSpec.Address, handlerInstruction, r)
+
+			resp, httpErr = handlerHTTPClient.Get(httpUri)
+
+			if httpErr != nil {
+				http.Error(w, httpErr.Error(), http.StatusServiceUnavailable)
+				log.Printf("HTTP GET error" + httpErr.Error())
+				return
+			}
+		} else if reqType == http.MethodPost {
+			httpUri := createUriFromRequest(*handlerEdgeDeviceSpec.Address, handlerInstruction, r)
+
+			requestBody, parseErr := io.ReadAll(r.Body)
+			if parseErr != nil {
+				log.Panic("Error on parsing body" + parseErr.Error())
```

Comment:
will the pod fail on parseErr ?

Reply:
yes, now changed to print instead of panic.